src/silx/gui/dialog/ColormapDialog.py

```diff
@@ -1723,9 +1736,28 @@ def _autoscaleModeUpdated(self):
 
         colormap = self.getColormap()
         if colormap is not None:
+            activate_saturation = self._autoScaleCombo.currentText() == "Percentile"
             with self._colormapChange:
+                if activate_saturation:
+                    colormap.setSaturation(self._saturation.value())
+                else:
+                    colormap.setSaturation(Colormap.DEFAULT_SATURATION)
                 colormap.setAutoscaleMode(mode)
 
+        self._updateSaturationVisibility()
+        self._updateWidgetRange()
+
+    def _updateSaturationVisibility(self):
```

Comment:
Why is this called at multiple places, can't it be connected directly to self._autoScaleCombo's currentTextChanged: https://doc.qt.io/qt-6/qcombobox.html#currentTextChanged